loadtest/control/browsercontroller/controller.go

```diff
@@ -0,0 +1,297 @@
+// by communicating with the browserLt API server.
+type BrowserController struct {
+	id              int
+	user            user.User
+	status          chan<- control.UserStatus
+	rate            float64
+	stopChan        chan struct{}
+	stoppedChan     chan struct{}
+	wg              *sync.WaitGroup
+	ltBrowserApiUrl string
+	httpClient      *http.Client
+	mu              sync.Mutex
+}
+
+// AddBrowserRequest represents the request body for adding a browser session
+type AddBrowserRequest struct {
+	// UserID is the username or email of the user to add a browser session
+	UserID   string `json:"userId" validate:"required"`
+	Password string `json:"password" validate:"required"`
+}
+
+// RemoveBrowserRequest represents the request body for removing a browser session
+type RemoveBrowserRequest struct {
+	// UserID is the username or email of the user to remove a browser session
+	UserID string `json:"userId" validate:"required"`
+}
+
+// BrowserAPIResponse represents the response from the browser API
+type BrowserAPIResponse struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+	Error   *struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+	} `json:"error,omitempty"`
+}
+
+// New creates and initializes a new BrowserController with given parameters.
+func New(id int, user user.User, status chan<- control.UserStatus) (*BrowserController, error) {
+	if user == nil {
+		return nil, errors.New("user cannot be nil")
+	}
+	if status == nil {
+		return nil, errors.New("status channel cannot be nil")
+	}
+
+	return &BrowserController{
+		id:              id,
+		user:            user,
+		status:          status,
+		rate:            1.0,
+		stopChan:        make(chan struct{}),
+		stoppedChan:     make(chan struct{}),
+		wg:              &sync.WaitGroup{},
+		ltBrowserApiUrl: LT_BROWSER_API_URL,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}, nil
+}
+
+// Run starts the browser controller by creating a browser session.
+// This corresponds to adding a user in the browser load test.
+func (c *BrowserController) Run() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.user == nil {
+		c.sendFailStatus("browser controller was not initialized")
+		return
+	}
+
+	c.status <- control.UserStatus{
+		ControllerId: c.id,
+		User:         c.user,
+		Info:         "browser controller started",
+		Code:         control.USER_STATUS_STARTED,
+	}
+
+	defer func() {
+		c.user.ClearUserData()
+		c.sendStopStatus()
+		close(c.stoppedChan)
+	}()
+
+	initActions := []control.UserAction{
+		control.SignUp,
+		control.Login,
+		control.JoinTeam,
+	}
+
+	for _, action := range initActions {
+		if resp := action(c.user); resp.Err != nil {
+			c.status <- c.newErrorStatus(resp.Err)
+			return
+		} else if resp.Info != "" {
+			c.status <- c.newInfoStatus(resp.Info)
+		}
+
+		// If stop signal is received then stop the controller
+		select {
+		case <-c.stopChan:
+			return
+		default:
+		}
+	}
```

Comment:
Shouldn't we do this via the browser itself?

Reply:
For now keeping it here is fine for simplicity. Once the browser simulation tests get more robust, we can move this logic there.